### PR TITLE
WatchProxyServer

### PR DESCRIPTION
**Pull request by bradyjoestar**
### What problem does this PR solve? 
#1478 

### What is changed and how it works?
add watch_stream and watchProxyServer

### Check List 

Tests 

 - Unit test: to be added.

@nolouch  
If the design should be changed, please tell me.

When the design is approved, I would remove the pd-client which is used for test.

:beers:

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -803,6 +807,55 @@ func (c *client) GetOperator(ctx context.Context, regionID uint64) (*pdpb.GetOpe
...
+func (c *client) Watch(ctx context.Context, key []byte) error {
+	cc := c.leaderClient()
+
+	stream, err := cc.Watch(context.Background())
+	if err != nil {
+		log.Error("failed to call: %v", zap.Error(err))
+		return err
+	}
+
+	if err != nil {
```
Comment (nolouch) [inline client/client.go]: duplicate?

File: client/client.go [hunk 2]
```diff
@@ -803,6 +807,55 @@ func (c *client) GetOperator(ctx context.Context, regionID uint64) (*pdpb.GetOpe
 	})
 }
 
+func (c *client) Watch(ctx context.Context, key []byte) error {
+	cc := c.leaderClient()
+
+	stream, err := cc.Watch(context.Background())
```
Comment (nolouch) [inline client/client.go]: use the outer ctx.

File: server/watch_stream.go [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+	}()
+}
+
+func (wps *WatchProxyServer) notifyAllObservers(key watchKey) {
+	for {
+		select {
+		// we heard a event from etcd and notify all observer
+		case watchResp := <-wps.watchers[key].watchChan:
+			for _, ws := range wps.watchers[key].watchStreams {
+				ws.stream.Send(&watchResp)
```
Comment (nolouch) [inline server/watch_stream.go]: when to delete the inactive stream?

**Comment by bradyjoestar** [comment]
@nolouch 
Thanks for your review! 
Recently I am a little busy, I would finish it in several days. :beer:

**Comment by bradyjoestar** [comment]
@nolouch 
I am recently focuing on `libos` in the `intel-sgx`, so sorry for delaying this pr.
I'd like to close this pr temporary. 

Sorry!
webb

**Comment by nolouch** [comment]
@bradyjoestar That's ok. Thanks.

**Pull request closed without merging**